tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -563,6 +563,14 @@ TILEDB_EXPORT int32_t tiledb_filestore_size(
   Array array(context, std::string(filestore_array_uri), TILEDB_READ);
 
   tiledb_datatype_t dtype;
+  if (!array.has_metadata(
+          tiledb::sm::constants::filestore_metadata_size_key, &dtype)) {
+    LOG_STATUS(Status_Error(
+        std::string("Filestore size key not found in array metadata; this "
+                    "array may not be a filestore: ") +
+        filestore_array_uri));
+    return TILEDB_ERR;
+  }
```

Comment:
Does this possibly leave user with a file that was stored 
(thought with some sort of error resulting in size not being stored)
that they now cannot retrieve (because a return is made)?
And if so, is there any way that this -could- be made to retrieve/return the data in spite of the missing filesize

Reply:
The export path uses/requires the same key, so I don't think so.